src/PhenX.EntityFrameworkCore.BulkInsert/Metadata/PropertyAccessor.cs

```diff
@@ -0,0 +1,66 @@
+using System.Reflection.Emit;
+using System.Reflection;
+
+namespace PhenX.EntityFrameworkCore.BulkInsert.Metadata;
+
+internal static class PropertyAccessor
+{
+    public delegate TValue Getter<TSource, TValue>(TSource source);
+
+    public static Getter<object, object?> CreateUntypedGetter(PropertyInfo propertyInfo, Type sourceType, Type valueType)
+    {
+        var method =
+            typeof(PropertyAccessor).GetMethod(nameof(CreateInternalUntypedGetter), BindingFlags.NonPublic)!
+                .MakeGenericMethod(sourceType, valueType);
+
+        return (Getter<object, object?>)method.Invoke(null, [propertyInfo])!;
+    }
+
+    private static Getter<object, object?> CreateInternalUntypedGetter<TSource, TValue>(PropertyInfo propertyInfo)
+    {
+        var getter = CreateGetter<TSource, TValue>(propertyInfo);
+
+        return source => getter((TSource)source!);
+    }
+
+    public static Getter<TSource, TValue> CreateGetter<TSource, TValue>(PropertyInfo propertyInfo)
+    {
+        if (!propertyInfo.CanRead)
+        {
+            return x => throw new NotSupportedException();
+        }
+
+        var bakingField =
```

Comment:
I'm curious about how much time and memory is won with thoses changes

Reply:
In the general context: Not that much because it is not a hot path, but reflection is relatively slow and I already had the code ;)